fem/libceed/ceed.cpp

```diff
@@ -575,7 +576,8 @@ void CeedAssembleDiagonalPA(const CeedData *ceedDataPtr,
    CeedScalar *d_ptr;
    CeedMemType mem;
    CeedGetPreferredMemType(internal::ceed, &mem);
-   if ( Device::Allows(Backend::CUDA) && mem==CEED_MEM_DEVICE )
+   if ( (Device::Allows(Backend::CUDA) || Device::Allows(Backend::HIP))
+        && mem==CEED_MEM_DEVICE )
```

Comment:
```suggestion
   if ( Device::Allows(Backend::DEVICE_MASK) && mem==CEED_MEM_DEVICE )
```

Reply:
These have been fixed in b267750431f09086aefa0494e2ab366ba2ef2620